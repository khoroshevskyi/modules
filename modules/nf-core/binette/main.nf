process BINETTE {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://depot.galaxyproject.org/singularity/binette:1.1.2--pyh7e72e81_0'
        : 'biocontainers/binette:1.1.2--pyh7e72e81_0'}"

    input:
    tuple val(meta), path(input_binning, stageAs: "input_binning/*"), path(contigs), path(proteins)
    val input_type
    path checkm2_db

    output:
    tuple val(meta), path("${meta.id}_final_bins/*")                  , emit: refined_bins
    tuple val(meta), path("${meta.id}_final_bins_quality_reports.tsv"), emit: refined_bins_report
    tuple val(meta), path("${meta.id}_input_bins_quality_reports/*")  , emit: input_bins_report
    path "versions.yml"                                               , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args        = task.ext.args ?: ''
    def prefix      = task.ext.prefix ?: "${meta.id}"
    def protein_arg = proteins ? "--proteins ${proteins}" : ""
    def input_arg   = ""

    if (input_type == 'fasta') {
        input_arg = "--bin_dirs"
    } else if (input_type == 'tsv') {
        input_arg = "--contig2bin_tables"
    } else {
        error "Invalid input_type: ${input_type}. Must be 'fasta' or 'tsv'"
    }

    """
    binette \\
        ${input_arg} input_binning/* \\
        ${protein_arg} \\
        --contigs ${contigs} \\
        --threads ${task.cpus} \\
        --checkm2_db ${checkm2_db} \\
        -o . \\
        ${args}

    mv final_bins "${prefix}_final_bins"
    mv final_bins_quality_reports.tsv ${prefix}_final_bins_quality_reports.tsv
    mv input_bins_quality_reports ${prefix}_input_bins_quality_reports

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        binette: \$( binette --version )
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    mkdir ${prefix}_final_bins
    touch ${prefix}_final_bins/bin_b.fasta

    mkdir ${prefix}_input_bins_quality_reports
    touch ${prefix}_input_bins_quality_reports/bin_a.tsv
    touch ${prefix}_final_bins_quality_reports.tsv


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        binette: \$( binette --version )
    END_VERSIONS
    """
}
