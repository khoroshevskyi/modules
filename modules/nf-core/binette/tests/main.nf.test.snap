{
    "Refine bins from TSV contig2bin tables": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "bin_3.fa:md5,2d036a992d3eb45d43fb1b17d9eb70e5"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test_final_bins_quality_reports.tsv:md5,29492f81a69b23e5d9aa261058552bdf"
                ]
            ],
            [
                "versions.yml:md5,a733c14a8248c5bd038523fd1a5ebc10"
            ],
            {
                "BINETTE": {
                    "binette": "1.1.2"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.1"
        },
        "timestamp": "2025-09-05T12:59:47.731509"
    },
    "test_binette - stub": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "bin_b.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test_final_bins_quality_reports.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "versions.yml:md5,a733c14a8248c5bd038523fd1a5ebc10"
            ],
            {
                "BINETTE": {
                    "binette": "1.1.2"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.1"
        },
        "timestamp": "2025-09-05T12:59:57.305202"
    },
    "Refine bins from FASTA input directories": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "bin_3.fa:md5,2d036a992d3eb45d43fb1b17d9eb70e5"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test_final_bins_quality_reports.tsv:md5,29492f81a69b23e5d9aa261058552bdf"
                ]
            ],
            [
                "versions.yml:md5,a733c14a8248c5bd038523fd1a5ebc10"
            ],
            {
                "BINETTE": {
                    "binette": "1.1.2"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.1"
        },
        "timestamp": "2025-09-05T12:59:25.446307"
    }
}